### ISSUE TITLE
47.14-DEV Princess NPE

### ISSUE THREAD

**Issue by gsparks3** [MegaMek/megamek#2400]
### Environment
MekHQ Integrated MegaMek
Windows 10
Java version 11.0.8

### Description
Princess stopped running during a match (and it sure is noticeable given how much faster she is now!)
Distinctive NPE error messages in both MegaMekLog.txt and MekHQlog.txt (see attached). 
Does not reproduce upon loading saved game, although save is still provided. 

### Files
Username: Greyscales
[princessNPE.sav.gz](https://github.com/MegaMek/megamek/files/5534506/princessNPE.sav.gz)
[megameklog.txt](https://github.com/MegaMek/megamek/files/5534484/megameklog.txt)
[mekhqlog.txt](https://github.com/MegaMek/megamek/files/5534493/mekhqlog.txt)

**Comment by HammerGS** [comment]
```
21:55:01,589 ERROR [megamek.client.bot.BotClient] {Capellan Confederation Turn 0 Calc Thread} 
calculateMyTurnWorker(), line 591 : Entity is null.
java.lang.NullPointerException: Entity is null.
	at megamek.client.bot.princess.Princess.continueMovementFor(Princess.java:1259)
	at megamek.client.bot.princess.Princess.calculateMoveTurn(Princess.java:1081)
	at megamek.client.bot.BotClient.calculateMyTurnWorker(BotClient.java:554)
	at megamek.client.bot.BotClient.calculateMyTurn(BotClient.java:516)
	at megamek.client.bot.BotClient.access$000(BotClient.java:85)
	at megamek.client.bot.BotClient$CalculateBotTurn.run(BotClient.java:109)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by NickAragua** [comment]
Did a unit get destroyed during the movement phase somehow?

**Comment by gsparks3** [comment]
The last thing that happened in the game log prior to saving and quitting is a Maxim Heavy Hover Transport ejecting (crew abandoned due to it being immobile). Not sure if that counts. 

MekHQlog attached above also has:

Exception in thread "Packet Pump" java.lang.NullPointerException
	at megamek.common.GameTurn.getClassCode(GameTurn.java:336)
	at megamek.common.GameTurn$EntityClassTurn.isValidEntity(GameTurn.java:387)
	at megamek.common.GameTurn.isValidEntity(GameTurn.java:74)
	at megamek.common.GameTurn.isValid(GameTurn.java:117)
	at megamek.server.Server.receiveMovement(Server.java:4896)
	at megamek.server.Server.handle(Server.java:31222)
	at megamek.server.Server$PacketPump.run(Server.java:259)

I think I've seen this sort of server hiccup before elsewhere. Not sure what causes it.

**Comment by NickAragua** [comment]
This is generally fixable by kicking and replacing the bot.

**Comment by gsparks3** [comment]
Had the same thing happen in 47.14. Interesting note: kicking and replacing the bot did NOT resolve the error (the bot's turn was still stalled). However, saving the game and reloading from the save did, so it still can't be reproduced for bugtesting ...

Attached save and logs. 
Playername: Greyscales
[princessServerNPE.sav.gz](https://github.com/MegaMek/megamek/files/5600638/princessServerNPE.sav.gz)
[megameklog.txt](https://github.com/MegaMek/megamek/files/5600621/megameklog.txt)
[mekhqlog.txt](https://github.com/MegaMek/megamek/files/5600622/mekhqlog.txt)

**Comment by Windchild292** [comment]
GameTurn::getClassCode NPE is fixed in #2683.

**Comment by gsparks3** [comment]
Nice! I was never able to reproduce this one reliably, so good to see that it got caught somewhere else.
